### DISCUSSION TITLE
Can the agent connect to the OAP server automatically again after the OAP server is recoveried?

### DISCUSSION THREAD

**Discussion by SimonLiuhd** [apache/skywalking#8636] [Q&A] [answered]
Hi all,
     The OAP server cluster crashed for some days, and it has been recoveried, but I see the storage (elasticsearch indices) contents are empty.  Before the cluster crashes, all things work well - there are datas showed on the sky walkingUI. 
    So, I am consufed, why it no data after its recoverying, cannot the agent connect to the OAP server automatically again? or must the application which is using the agent be restarted?

Thanks a lot.

**Comment by wu-sheng** [comment, marked as answer]
> cannot the agent connect to the OAP server automatically again? 

The agent will reconnect automatically.

> or must the application which is using the agent be restarted?

No, you don't have to do so.

**Comment by SimonLiuhd** [comment]
Thanks @wu-sheng 
I'm using the old version 8.2. 
From which version the feature that the agent will reconnect automatically?

**Reply by wu-sheng** [reply]
Always from v6? But you should check change logs, maybe some versions have bugs. You should know 8.2 is a very old release, one year ago?
